---
title: "An introduction to the `ASCETIC` R package"
author: 
  - Luca De Sano
  - Daniele Ramazzotti
date: "`r format(Sys.time(), '%B %d, %Y')`"
graphics: yes
package: ASCETIC
output:
  BiocStyle::html_document:
    toc_float: true
vignette: >
  %\VignetteIndexEntry{An R package for the inference of statistically significant temporal patterns involving alternations in driver genes from cancer genomics data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
  %\VignetteDepends{ASCETIC,BiocStyle}
---

Cancer development is a stochastic process involving large populations of cells. Random genetic and epigenetic alterations commonly occurring in any cell can occasionally be beneficial to neoplastic ones, thus defining clones characterized by a functional selective advantage. During clonal evolution, certain clones can be positively selected for increased proliferation and survival ability, outgrowing competing cells and this can eventually lead to invasion and metastasis. Throughout such a multi-step stochastic process, cancer cells can acquire over time a set of biological capabilities that sometimes are referred to as hallmarks. Not all variants are involved in their acquisition, but only a relative small subset of them – i.e., the drivers –, while most mutations present in the cancer clones do not increase their fitness – i.e., the passengers.

In response to the constantly increasing availability of cancer omics data and the rapid advancement of data science and machine learning techniques, we introduce a novel framework named ASCETIC (Agony-baSed Cancer EvoluTion InferenCe, https://www.nature.com/articles/s41467-023-41670-3). ASCETIC can extract cancer's evolutionary signatures, which represent recurring paths of driver mutation acquisition associated with distinct disease outcomes. ASCETIC can process sequencing data derived from different technologies, including bulk and single-cell sequencing. The approach goes beyond the traditional focus on individual genetic alterations and explores the broader landscape of genomic alterations and their intricate interactions, with the aim to enhance predictive accuracy and our understanding of cancer evolution's impact on prognosis.

ASCETIC's workflow involves the reconstruction of robust tumor evolution models for individual patients, followed by the integration of these models into a comprehensive cancer-specific evolution model. By leveraging prognostic data through regularized Cox regression, ASCETIC identifies significant evolutionary patterns or signatures associated with patient outcomes.

In summary, ASCETIC represents a powerful tool for uncovering consistent evolutionary patterns in cancer, offering the potential to contribute to a curated catalogue of evolutionary signatures akin to the widely used COSMIC Mutational Signatures database.

In its basic implementation, ASCETIC requires two main inputs: (i) a binary matrix where rows are patients (i.e., samples) and columns are mutations. Each cell of the matrix is 1 if the related mutation was observed in the sample; 0 otherwise. (2) Information of temporal patters across the considered driver genes. Such information can be determined either from classical NGS genomics data considering a single biopsy per patient, in terms of cancer cell fractions reported for each gene, or from data at different resolutions such as multi-region or single-cell data where multiple samples per patients are provided. In this latter case, ASCETIC takes as input a phylogenetic model per patient, represented as a mutational tree. Such trees where each node corresponds to a mutation, display the mutation history of a tumor.

The ASCETIC framework is built upon the observation that, in most cases, the accumulation of passenger mutations during cancer progression follows random dynamics. However, a small set of genes drive tumor evolution, and for these alterations, drift-driven evolution and selective pressures may lead to a consistent ordering across multiple patients. However, this ordering may not be unique and can be confounded by the presence of heterogeneous cancer subtypes within a tumor dataset. To address this challenge, ASCETIC decomposes the inference problem into three main tasks.

First, it combines information of temporal patters obtained for multiple patients in order to create an agony-derived ranking of the considered alterations, providing a partial temporal ordering among the drivers during cancer evolution. Agony is a measure of hierarchy within a directed graph. Given a directed graph and a ranking metric (e.g., in our case the time ordering of accumulation of driver alterations during tumor evolution), any arc from nodes that are higher in the hierarchy (e.g., alterations that occur in later stages of the tumor) to nodes that are lower in the hierarchy (e.g., alterations that occur at the initiation of the tumor) are not expected and they are said to be causing agony. The output of this phase is a partially ordered set (poset), which minimizes inconsistencies measured in terms of agony.

Then, the approach adopts a likelihood-based approach grounded in the theory of probabilistic causation to select the most significant relationships among driver genes in a Bayesian Network that depicts repeated evolutionary trajectories. Finally, ASCETIC leverages the inferred evolutionary steps to stratify patients based on survival data and selects the most relevant features from the model to cluster the samples into different risk groups or clusters. Survival analysis of the different risk groups is then performed via the standard Kaplan-Meier estimate. ASCETIC outputs a set of evolutionary signatures associated with the different risk groups, displaying the inferred risk for each of the selected evolutionary steps and their relative prevalence in the cluster.

In this vignette, we give an overview of the package by presenting some of its main functions.

# Installing the ASCETIC R package

The ASCETIC package can be installed from Bioconductor as follows.

```{r eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ASCETIC")
```

Or it can be installed directly from GitHub using the R package devtools as follows.

```{r eval=FALSE}
library("devtools")
install_github("danro9685/ASCETIC", ref = 'master')
```

# Changelog

- *0.99.20* Package pre-release in October 2023.
- *0.99.0* Package created in March 2023.

# Using the ASCETIC R package

We provide within the package example datasets for both the single biopsy and the multi-regions/single-cell cases. We now load such example datasets.

```{r}
library("ASCETIC")
data(datasetExampleSingleSamples)
data(ccfDatasetExampleSingleSamples)
data(datasetExamplePhylogenies)
data(modelsPhylogenies)
```

First, we outline the inference steps for the single biopsy NGS data scenario, where only one sample per patient is available and the temporal patters among genes are estimated considering cancer cell fractions data. To test this, we consider an example dataset comprising 10 hepatocellular carcinoma sequencing samples from Nguyen, Bastien, et al., 2022. We recall that such (reduced and partial) dataset was provided only for computational testing and should not be used for biological analysis/conclusions.

We now perform the basic ASCETIC inference.

```{r}
set.seed(12345)
resExampleSingleSamples <- asceticCCF(
                              dataset = datasetExampleSingleSamples,
                              ccfDataset = ccfDatasetExampleSingleSamples,
                              regularization = "aic",
                              command = "hc", 
                              restarts = 0 )
```

The output of this analysis is a a list of 4 elements: 1) dataset, input dataset; 2) ccfDataset, input ccf dataset; 3) poset, partially order set among mutations estimated by ASCETIC from the agony ranking and 4) inference, inferred ASCETIC evolutionary model for each selected regularization.

The poset represents an estimate of the agony ranking for this dataset, while the Bayesian Network obtained after the likelihood-fit step is the inference output of ASCETIC, which provides the inferred repeated evolutionary trajectories.

To improve the statistical stability of the framework expecially in the common situation of noisy data, ASCETIC implements a re-sampling algorithm which builds on the function just described. This procedure is more computational intensive compared to the basic one, but should be preferred on real data. It requires an additional dataset with variant allele frequencies information and can be executed as follows.

```{r}
set.seed(12345)
data(vafDatasetExampleSingleSamples)
resExampleSingleSamplesResampling <- asceticCCFResampling(
                                              dataset = datasetExampleSingleSamples,
                                              ccfDataset = ccfDatasetExampleSingleSamples,
                                              vafDataset = vafDatasetExampleSingleSamples,
                                              nsampling = 5,
                                              regularization = "aic",
                                              command = "hc",
                                              restarts = 0 )
```

We refer to the manual for a detailed description of each parameter.

We now describe the execution of the ASCETIC framework when high-resolution cancer genomics data are available, such as multi-regions data or single-cell data. In this case, phylogenetic reconstructions for each patient are available and are given as input to ASCETIC as a adjacency matrices representing mutational trees. We consider an other example dataset consisting of 10 myeloid malignancies samples sequenced with Tapestri from Miles, Linde A., et al., 2020. We recall that also in this case, such (reduced and partial) dataset was provided only for computational testing and should not be used for biological analysis/conclusions.

We now perform the basic ASCETIC inference for this second scenario.

```{r}
set.seed(12345)
resExamplePhylogeniesDataset <- asceticPhylogenies(
                                      dataset = datasetExamplePhylogenies,
                                      models = modelsPhylogenies,
                                      regularization = "aic",
                                      command = "hc",
                                      restarts = 0)
```

The output of this analysis is a a list of 4 elements: 1) dataset, input dataset; 2) models, input mutational trees; 3) poset, partially order set among mutations estimated by ASCETIC from the agony ranking and 4) inference, inferred ASCETIC evolutionary model for each selected regularization.

The resulting poset and inference elements have the same semantics as the previous example.

Also in this case we provide a method to improve the statistical stability of the framework expecially in the situation of noisy data. ASCETIC implements a bootstrap procedure which builds on the function just described. This procedure should be preferred on real data. It can be executed as follows.

```{r}
set.seed(12345)
resExamplePhylogeniesDatasetBootstrap <- asceticPhylogeniesBootstrap(
                                                   dataset = datasetExamplePhylogenies,
                                                   models = modelsPhylogenies,
                                                   nsampling = 5,
                                                   regularization = "aic",
                                                   command = "hc",
                                                   restarts = 0 )
```

ASCETIC allows for estimating confidence in the inferred evolutionary trajectories through cross-validation. Specifically, the algorithm is applied multiple times on sampled data (80% sampling), and statistics are returned regarding the confidence of each evolutionary step.

Cross-validation can be executed for both bulk data providing single samples and for multi-region bulk data and single-cell data. The estimation can be executed as follows.

```{r}
resExampleCCFAssessment <- asceticCCFAssessment(
                                       inference = resExampleSingleSamplesResampling,
                                       niterations = 5,
                                       vafDataset = vafDatasetExampleSingleSamples,
                                       nsampling = 5)
```

```{r}
set.seed(12345)
resExamplePhylogeniesAssessment <- asceticPhylogeniesAssessment(
                                          inference = resExamplePhylogeniesDatasetBootstrap,
                                          niterations = 5,
                                          nsampling = 5)
```

Finally, ASCETIC can perform the extraction of evolutionary signatures significantly associated to prognosis and use them to stratify patients. This can be done as follows.

```{r}
set.seed(12345)
data(amlExample)
resExampleEvosigs <- evoSigs(survivalData = amlExample$survival_data, 
                             evolutionarySteps = amlExample$evolutionary_steps)
print(names(resExampleEvosigs))
print(resExampleEvosigs$evolutionarySteps)
print(resExampleEvosigs$clustersPrevalence)
```

The accumulation of alterations in driver genes can follow repeated routes in different cancer patients. Detecting such trajectories could be important in order to implement appropriate therapeutic responses. As a matter of fact, being able to stratify cancer patients based on their molecular evolution could enable the prediction of the future steps of the disease progression, potentially allowing the execution of optimal and personalized treatments that anticipate the next stages of the cancer's evolution. In our example, ASCETIC could identify a set of evolutionary steps (see evolutionarySteps and clustersPrevalence variables) with a significant association to prognosis.

Specifically, the evolutionarySteps variable reports 20 evolutionary steps that, in this example dataset, show a significant association with overall survival. The steps with negative values are negatively associated with risk, meaning that patients with these mutations have a better prognosis. On the other hand, steps with positive values increase the risk, indicating that patients with these mutations have a worse prognosis.

Finally, the clustersPrevalence variable provides the three clusters discovered based on the detected evolutionary signatures, along with the relative prevalences of each evolutionary step in the clusters.

We refer to the manual for a detailed description of each parameter and to the ASCETIC manuscript for details on the method.

# Current R Session

```{r}
sessionInfo()
```
