<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using the package â€¢ ASCETIC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using the package">
<meta property="og:description" content="ASCETIC">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ASCETIC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_introduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/2_using_the_package.html">Using the package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/danro9685/ASCETIC/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="2_using_the_package_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using the package</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/danro9685/ASCETIC/blob/HEAD/vignettes/2_using_the_package.rmd" class="external-link"><code>vignettes/2_using_the_package.rmd</code></a></small>
      <div class="hidden name"><code>2_using_the_package.rmd</code></div>

    </div>

    
    
<p>We provide within the package example datasets for both the single biopsy and the multi-regions/single-cell cases. We now load such example datasets.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/danro9685/ASCETIC" class="external-link">"ASCETIC"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dataset_example_single_samples</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ccf_dataset_example_single_samples</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dataset_example_phylogenies</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">models_phylogenies</span><span class="op">)</span></span></code></pre></div>
<p>First, we outline the inference steps for the single biopsy NGS data scenario, where only one sample per patient is available and the temporal patters among genes are estimated considering cancer cell fractions data. To test this, we consider an example dataset comprising 10 hepatocellular carcinoma sequencing samples from Nguyen, Bastien, et al., 2022. We recall that such (reduced and partial) dataset was provided only for computational testing and should not be used for biological analysis/conclusions.</p>
<p>We now perform the basic ASCETIC inference.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">res_example_single_samples</span> <span class="op">=</span> <span class="fu"><a href="../reference/perform.ascetic.ccf.dataset.html">perform.ascetic.ccf.dataset</a></span><span class="op">(</span> </span>
<span>                                dataset <span class="op">=</span> <span class="va">dataset_example_single_samples</span>, </span>
<span>                                ccf.dataset <span class="op">=</span> <span class="va">ccf_dataset_example_single_samples</span>, </span>
<span>                                regularization <span class="op">=</span> <span class="st">"aic"</span>, </span>
<span>                                command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                                restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2 3</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 3 2</span></span></code></pre>
<p>The output of this analysis is a a list of 4 elements: 1) dataset, input dataset; 2) ccf dataset, input ccf dataset; 3) poset, partially order set among mutations estimated by ASCETIC from the agony ranking and 4) inference, inferred ASCETIC evolutionary model for each selected regularization.</p>
<p>To improve the statistical stability of the framework expecially in the common situation of noisy data, ASCETIC implements a re-sampling algorithm which builds on the function just described. This procedure is more computational intensive compared to the basic one, but should be preferred on real data. It requires an additional dataset with variant allele frequencies information and can be executed as follow.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">vaf_dataset_example_single_samples</span><span class="op">)</span></span>
<span><span class="va">res_example_single_samples_resampling</span> <span class="op">=</span> <span class="fu"><a href="../reference/perform.ascetic.ccf.dataset.resampling.html">perform.ascetic.ccf.dataset.resampling</a></span><span class="op">(</span> </span>
<span>                                            dataset <span class="op">=</span> <span class="va">dataset_example_single_samples</span>, </span>
<span>                                            ccf.dataset <span class="op">=</span> <span class="va">ccf_dataset_example_single_samples</span>, </span>
<span>                                            vaf.dataset <span class="op">=</span> <span class="va">vaf_dataset_example_single_samples</span>, </span>
<span>                                            nsampling <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                                            regularization <span class="op">=</span> <span class="st">"aic"</span>, </span>
<span>                                            command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                                            restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 0 </span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 0.2 </span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 0.4 </span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 0.6 </span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 0.8 </span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 3 2</span></span>
<span><span class="co">## 2 1</span></span>
<span><span class="co">## 2 3</span></span>
<span><span class="co">## 1</span></span></code></pre>
<p>We refer to the manual for a detailed description of each parameter.</p>
<p>We now describe the execution of the ASCETIC framework when high-resolution cancer genomics data are available, such as multi-regions data or single-cell data. In this case, phylogenetic reconstructions for each patient are available and are given as input to ASCETIC. We consider an other example dataset consisting of 10 myeloid malignancies samples sequenced with Tapestri from Miles, Linde A., et al., 2020. We recall that also in this case, such (reduced and partial) dataset was provided only for computational testing and should not be used for biological analysis/conclusions.</p>
<p>We now perform the basic ASCETIC inference for this second scenario.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">res_example_phylogenies_dataset</span> <span class="op">=</span> <span class="fu"><a href="../reference/perform.ascetic.phylogenies.dataset.html">perform.ascetic.phylogenies.dataset</a></span><span class="op">(</span> </span>
<span>                                    dataset <span class="op">=</span> <span class="va">dataset_example_phylogenies</span>, </span>
<span>                                    models <span class="op">=</span> <span class="va">models_phylogenies</span>, </span>
<span>                                    regularization <span class="op">=</span> <span class="st">"aic"</span>, </span>
<span>                                    command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                                    restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 16 1</span></span>
<span><span class="co">## 25 7</span></span>
<span><span class="co">## 7 12</span></span>
<span><span class="co">## 12 14</span></span>
<span><span class="co">## 12 16</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 8 8</span></span>
<span><span class="co">## 8 16</span></span>
<span><span class="co">## 26 1</span></span>
<span><span class="co">## 26 25</span></span>
<span><span class="co">## 12 26</span></span>
<span><span class="co">## 22 8</span></span>
<span><span class="co">## 5 9</span></span>
<span><span class="co">## 5 22</span></span>
<span><span class="co">## 21 7</span></span>
<span><span class="co">## 16 14</span></span>
<span><span class="co">## 25 16</span></span>
<span><span class="co">## 7 19</span></span>
<span><span class="co">## 14 21</span></span>
<span><span class="co">## 12 4</span></span>
<span><span class="co">## 12 24</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 9 9</span></span>
<span><span class="co">## 9 11</span></span>
<span><span class="co">## 11 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 7 6</span></span>
<span><span class="co">## 7 7</span></span>
<span><span class="co">## 7 24</span></span>
<span><span class="co">## 7 26</span></span>
<span><span class="co">## 7 1</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 23 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 2 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 3 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 10 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 13 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 17 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 18 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 20 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 27 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<p>The output of this analysis is a a list of 4 elements: 1) dataset, input dataset; 2) models, input phylogenetic models; 3) poset, partially order set among mutations estimated by ASCETIC from the agony ranking and 4) inference, inferred ASCETIC evolutionary model for each selected regularization.</p>
<p>Also in this case we provide a method to improve the statistical stability of the framework expecially in the situation of noisy data. ASCETIC implements a bootstrap procedure which builds on the function just described. This procedure should be preferred on real data. It can be executed as follow.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">res_example_phylogenies_dataset_bootstrap</span> <span class="op">=</span> <span class="fu"><a href="../reference/perform.ascetic.phylogenies.dataset.bootstrap.html">perform.ascetic.phylogenies.dataset.bootstrap</a></span><span class="op">(</span> </span>
<span>                                                dataset <span class="op">=</span> <span class="va">dataset_example_phylogenies</span>, </span>
<span>                                                models <span class="op">=</span> <span class="va">models_phylogenies</span>, </span>
<span>                                                nsampling <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                                                regularization <span class="op">=</span> <span class="st">"aic"</span>, </span>
<span>                                                command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                                                restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 0 </span></span>
<span><span class="co">## 26 1</span></span>
<span><span class="co">## 26 25</span></span>
<span><span class="co">## 12 26</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span>
<span><span class="co">## 7 6</span></span>
<span><span class="co">## 7 7</span></span>
<span><span class="co">## 7 24</span></span>
<span><span class="co">## 7 26</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span>
<span><span class="co">## 7 6</span></span>
<span><span class="co">## 7 7</span></span>
<span><span class="co">## 7 24</span></span>
<span><span class="co">## 7 26</span></span>
<span><span class="co">## 8 8</span></span>
<span><span class="co">## 8 16</span></span>
<span><span class="co">## 12 4</span></span>
<span><span class="co">## 12 24</span></span>
<span><span class="co">## 12 4</span></span>
<span><span class="co">## 12 24</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 9 9</span></span>
<span><span class="co">## 9 11</span></span>
<span><span class="co">## 11 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span>
<span><span class="co">## 0.2 </span></span>
<span><span class="co">## 16 1</span></span>
<span><span class="co">## 25 7</span></span>
<span><span class="co">## 7 12</span></span>
<span><span class="co">## 12 14</span></span>
<span><span class="co">## 12 16</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 7 6</span></span>
<span><span class="co">## 7 7</span></span>
<span><span class="co">## 7 24</span></span>
<span><span class="co">## 7 26</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 9 9</span></span>
<span><span class="co">## 9 11</span></span>
<span><span class="co">## 11 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 12 4</span></span>
<span><span class="co">## 12 24</span></span>
<span><span class="co">## 16 1</span></span>
<span><span class="co">## 25 7</span></span>
<span><span class="co">## 7 12</span></span>
<span><span class="co">## 12 14</span></span>
<span><span class="co">## 12 16</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 22 8</span></span>
<span><span class="co">## 5 9</span></span>
<span><span class="co">## 5 22</span></span>
<span><span class="co">## 7 6</span></span>
<span><span class="co">## 7 7</span></span>
<span><span class="co">## 7 24</span></span>
<span><span class="co">## 7 26</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span>
<span><span class="co">## 26 1</span></span>
<span><span class="co">## 26 25</span></span>
<span><span class="co">## 12 26</span></span>
<span><span class="co">## 7 1</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 23 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 0.4 </span></span>
<span><span class="co">## 22 8</span></span>
<span><span class="co">## 5 9</span></span>
<span><span class="co">## 5 22</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 9 9</span></span>
<span><span class="co">## 9 11</span></span>
<span><span class="co">## 11 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 8 8</span></span>
<span><span class="co">## 8 16</span></span>
<span><span class="co">## 22 8</span></span>
<span><span class="co">## 5 9</span></span>
<span><span class="co">## 5 22</span></span>
<span><span class="co">## 7 1</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 23 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 7 1</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 23 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 22 8</span></span>
<span><span class="co">## 5 9</span></span>
<span><span class="co">## 5 22</span></span>
<span><span class="co">## 7 6</span></span>
<span><span class="co">## 7 7</span></span>
<span><span class="co">## 7 24</span></span>
<span><span class="co">## 7 26</span></span>
<span><span class="co">## 12 4</span></span>
<span><span class="co">## 12 24</span></span>
<span><span class="co">## 0.6 </span></span>
<span><span class="co">## 7 1</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 23 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 21 7</span></span>
<span><span class="co">## 16 14</span></span>
<span><span class="co">## 25 16</span></span>
<span><span class="co">## 7 19</span></span>
<span><span class="co">## 14 21</span></span>
<span><span class="co">## 26 1</span></span>
<span><span class="co">## 26 25</span></span>
<span><span class="co">## 12 26</span></span>
<span><span class="co">## 16 1</span></span>
<span><span class="co">## 25 7</span></span>
<span><span class="co">## 7 12</span></span>
<span><span class="co">## 12 14</span></span>
<span><span class="co">## 12 16</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 16 1</span></span>
<span><span class="co">## 25 7</span></span>
<span><span class="co">## 7 12</span></span>
<span><span class="co">## 12 14</span></span>
<span><span class="co">## 12 16</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 21 7</span></span>
<span><span class="co">## 16 14</span></span>
<span><span class="co">## 25 16</span></span>
<span><span class="co">## 7 19</span></span>
<span><span class="co">## 14 21</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span>
<span><span class="co">## 26 1</span></span>
<span><span class="co">## 26 25</span></span>
<span><span class="co">## 12 26</span></span>
<span><span class="co">## 26 1</span></span>
<span><span class="co">## 26 25</span></span>
<span><span class="co">## 12 26</span></span>
<span><span class="co">## 0.8 </span></span>
<span><span class="co">## 26 1</span></span>
<span><span class="co">## 26 25</span></span>
<span><span class="co">## 12 26</span></span>
<span><span class="co">## 16 1</span></span>
<span><span class="co">## 25 7</span></span>
<span><span class="co">## 7 12</span></span>
<span><span class="co">## 12 14</span></span>
<span><span class="co">## 12 16</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 7 15</span></span>
<span><span class="co">## 15 28</span></span>
<span><span class="co">## 7 6</span></span>
<span><span class="co">## 7 7</span></span>
<span><span class="co">## 7 24</span></span>
<span><span class="co">## 7 26</span></span>
<span><span class="co">## 7 1</span></span>
<span><span class="co">## 15 7</span></span>
<span><span class="co">## 23 15</span></span>
<span><span class="co">## 7 25</span></span>
<span><span class="co">## 7 6</span></span>
<span><span class="co">## 7 7</span></span>
<span><span class="co">## 7 24</span></span>
<span><span class="co">## 7 26</span></span>
<span><span class="co">## 22 8</span></span>
<span><span class="co">## 5 9</span></span>
<span><span class="co">## 5 22</span></span>
<span><span class="co">## 8 8</span></span>
<span><span class="co">## 8 16</span></span>
<span><span class="co">## 26 1</span></span>
<span><span class="co">## 26 25</span></span>
<span><span class="co">## 12 26</span></span>
<span><span class="co">## 12 4</span></span>
<span><span class="co">## 12 24</span></span>
<span><span class="co">## 1</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 2 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 3 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 10 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 13 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 17 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 18 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 20 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<pre><code><span><span class="co">## Warning in check.data(x): variable 27 has levels that are not observed in the</span></span>
<span><span class="co">## data.</span></span></code></pre>
<p>We refer to the manual for a detailed description of each parameter and to the ASCETIC manuscript for details on the method.</p>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Daniele Ramazzotti, Luca De Sano.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
