<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to the `ASCETIC` R package • ASCETIC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to the `ASCETIC` R package">
<meta property="og:description" content="ASCETIC">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ASCETIC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/ASCETIC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/danro9685/ASCETIC/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to the <code>ASCETIC</code> R
package</h1>
                        <h4 data-toc-skip class="author">Luca De
Sano</h4>
                        <h4 data-toc-skip class="author">Daniele
Ramazzotti</h4>
            
            <h4 data-toc-skip class="date">October 31, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/danro9685/ASCETIC/blob/HEAD/vignettes/ASCETIC.rmd" class="external-link"><code>vignettes/ASCETIC.rmd</code></a></small>
      <div class="hidden name"><code>ASCETIC.rmd</code></div>

    </div>

    
    
<p>Cancer development is a stochastic process involving large
populations of cells. Random genetic and epigenetic alterations commonly
occurring in any cell can occasionally be beneficial to neoplastic ones,
thus defining clones characterized by a functional selective advantage.
During clonal evolution, certain clones can be positively selected for
increased proliferation and survival ability, outgrowing competing cells
and this can eventually lead to invasion and metastasis. Throughout such
a multi-step stochastic process, cancer cells can acquire over time a
set of biological capabilities that sometimes are referred to as
hallmarks. Not all variants are involved in their acquisition, but only
a relative small subset of them – i.e., the drivers –, while most
mutations present in the cancer clones do not increase their fitness –
i.e., the passengers.</p>
<p>In response to the constantly increasing availability of cancer omics
data and the rapid advancement of data science and machine learning
techniques, we introduce a novel framework named ASCETIC (Agony-baSed
Cancer EvoluTion InferenCe, <a href="https://www.nature.com/articles/s41467-023-41670-3" class="external-link uri">https://www.nature.com/articles/s41467-023-41670-3</a>).
ASCETIC can extract cancer’s evolutionary signatures, which represent
recurring paths of driver mutation acquisition associated with distinct
disease outcomes. ASCETIC can process sequencing data derived from
different technologies, including bulk and single-cell sequencing. The
approach goes beyond the traditional focus on individual genetic
alterations and explores the broader landscape of genomic alterations
and their intricate interactions, with the aim to enhance predictive
accuracy and our understanding of cancer evolution’s impact on
prognosis.</p>
<p>ASCETIC’s workflow involves the reconstruction of robust tumor
evolution models for individual patients, followed by the integration of
these models into a comprehensive cancer-specific evolution model. By
leveraging prognostic data through regularized Cox regression, ASCETIC
identifies significant evolutionary patterns or signatures associated
with patient outcomes.</p>
<p>In summary, ASCETIC represents a powerful tool for uncovering
consistent evolutionary patterns in cancer, offering the potential to
contribute to a curated catalogue of evolutionary signatures akin to the
widely used COSMIC Mutational Signatures database.</p>
<p>In its basic implementation, ASCETIC requires two main inputs: (i) a
binary matrix where rows are patients (i.e., samples) and columns are
mutations. Each cell of the matrix is 1 if the related mutation was
observed in the sample; 0 otherwise. (2) Information of temporal patters
across the considered driver genes. Such information can be determined
either from classical NGS genomics data considering a single biopsy per
patient, in terms of cancer cell fractions reported for each gene, or
from data at different resolutions such as multi-region or single-cell
data where multiple samples per patients are provided. In this latter
case, ASCETIC takes as input a phylogenetic model per patient,
represented as a mutational tree. Such trees where each node corresponds
to a mutation, display the mutation history of a tumor.</p>
<p>The ASCETIC framework is built upon the observation that, in most
cases, the accumulation of passenger mutations during cancer progression
follows random dynamics. However, a small set of genes drive tumor
evolution, and for these alterations, drift-driven evolution and
selective pressures may lead to a consistent ordering across multiple
patients. However, this ordering may not be unique and can be confounded
by the presence of heterogeneous cancer subtypes within a tumor dataset.
To address this challenge, ASCETIC decomposes the inference problem into
three main tasks.</p>
<p>First, it combines information of temporal patters obtained for
multiple patients in order to create an agony-derived ranking of the
considered alterations, providing a partial temporal ordering among the
drivers during cancer evolution. Agony is a measure of hierarchy within
a directed graph. Given a directed graph and a ranking metric (e.g., in
our case the time ordering of accumulation of driver alterations during
tumor evolution), any arc from nodes that are higher in the hierarchy
(e.g., alterations that occur in later stages of the tumor) to nodes
that are lower in the hierarchy (e.g., alterations that occur at the
initiation of the tumor) are not expected and they are said to be
causing agony. The output of this phase is a partially ordered set
(poset), which minimizes inconsistencies measured in terms of agony.</p>
<p>Then, the approach adopts a likelihood-based approach grounded in the
theory of probabilistic causation to select the most significant
relationships among driver genes in a Bayesian Network that depicts
repeated evolutionary trajectories. Finally, ASCETIC leverages the
inferred evolutionary steps to stratify patients based on survival data
and selects the most relevant features from the model to cluster the
samples into different risk groups or clusters. Survival analysis of the
different risk groups is then performed via the standard Kaplan-Meier
estimate. ASCETIC outputs a set of evolutionary signatures associated
with the different risk groups, displaying the inferred risk for each of
the selected evolutionary steps and their relative prevalence in the
cluster.</p>
<p>In this vignette, we give an overview of the package by presenting
some of its main functions.</p>
<div class="section level2">
<h2 id="installing-the-ascetic-r-package">Installing the ASCETIC R package<a class="anchor" aria-label="anchor" href="#installing-the-ascetic-r-package"></a>
</h2>
<p>The ASCETIC package can be installed from Bioconductor as
follows.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"ASCETIC"</span><span class="op">)</span></span></code></pre></div>
<p>Or it can be installed directly from GitHub using the R package
devtools as follows.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"danro9685/ASCETIC"</span>, ref <span class="op">=</span> <span class="st">'master'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="changelog">Changelog<a class="anchor" aria-label="anchor" href="#changelog"></a>
</h2>
<ul>
<li>
<em>0.99.20</em> Package pre-release in October 2023.</li>
<li>
<em>0.99.0</em> Package created in March 2023.</li>
</ul>
</div>
<div class="section level2">
<h2 id="using-the-ascetic-r-package">Using the ASCETIC R package<a class="anchor" aria-label="anchor" href="#using-the-ascetic-r-package"></a>
</h2>
<p>We provide within the package example datasets for both the single
biopsy and the multi-regions/single-cell cases. We now load such example
datasets.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/danro9685/ASCETIC" class="external-link">"ASCETIC"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">datasetExampleSingleSamples</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ccfDatasetExampleSingleSamples</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">datasetExamplePhylogenies</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">modelsPhylogenies</span><span class="op">)</span></span></code></pre></div>
<p>First, we outline the inference steps for the single biopsy NGS data
scenario, where only one sample per patient is available and the
temporal patters among genes are estimated considering cancer cell
fractions data. To test this, we consider an example dataset comprising
10 hepatocellular carcinoma sequencing samples from Nguyen, Bastien, et
al., 2022. We recall that such (reduced and partial) dataset was
provided only for computational testing and should not be used for
biological analysis/conclusions.</p>
<p>We now perform the basic ASCETIC inference.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">resExampleSingleSamples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/asceticCCF.html">asceticCCF</a></span><span class="op">(</span></span>
<span>                              dataset <span class="op">=</span> <span class="va">datasetExampleSingleSamples</span>,</span>
<span>                              ccfDataset <span class="op">=</span> <span class="va">ccfDatasetExampleSingleSamples</span>,</span>
<span>                              regularization <span class="op">=</span> <span class="st">"aic"</span>,</span>
<span>                              command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                              restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<p>The output of this analysis is a a list of 4 elements: 1) dataset,
input dataset; 2) ccfDataset, input ccf dataset; 3) poset, partially
order set among mutations estimated by ASCETIC from the agony ranking
and 4) inference, inferred ASCETIC evolutionary model for each selected
regularization.</p>
<p>The poset represents an estimate of the agony ranking for this
dataset, while the Bayesian Network obtained after the likelihood-fit
step is the inference output of ASCETIC, which provides the inferred
repeated evolutionary trajectories.</p>
<p>To improve the statistical stability of the framework expecially in
the common situation of noisy data, ASCETIC implements a re-sampling
algorithm which builds on the function just described. This procedure is
more computational intensive compared to the basic one, but should be
preferred on real data. It requires an additional dataset with variant
allele frequencies information and can be executed as follows.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">vafDatasetExampleSingleSamples</span><span class="op">)</span></span>
<span><span class="va">resExampleSingleSamplesResampling</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/asceticCCFResampling.html">asceticCCFResampling</a></span><span class="op">(</span></span>
<span>                                              dataset <span class="op">=</span> <span class="va">datasetExampleSingleSamples</span>,</span>
<span>                                              ccfDataset <span class="op">=</span> <span class="va">ccfDatasetExampleSingleSamples</span>,</span>
<span>                                              vafDataset <span class="op">=</span> <span class="va">vafDatasetExampleSingleSamples</span>,</span>
<span>                                              nsampling <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                                              regularization <span class="op">=</span> <span class="st">"aic"</span>,</span>
<span>                                              command <span class="op">=</span> <span class="st">"hc"</span>,</span>
<span>                                              restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 0 </span></span>
<span><span class="co">## 0.2 </span></span>
<span><span class="co">## 0.4 </span></span>
<span><span class="co">## 0.6 </span></span>
<span><span class="co">## 0.8 </span></span>
<span><span class="co">## 1</span></span></code></pre>
<p>We refer to the manual for a detailed description of each
parameter.</p>
<p>We now describe the execution of the ASCETIC framework when
high-resolution cancer genomics data are available, such as
multi-regions data or single-cell data. In this case, phylogenetic
reconstructions for each patient are available and are given as input to
ASCETIC as a adjacency matrices representing mutational trees. We
consider an other example dataset consisting of 10 myeloid malignancies
samples sequenced with Tapestri from Miles, Linde A., et al., 2020. We
recall that also in this case, such (reduced and partial) dataset was
provided only for computational testing and should not be used for
biological analysis/conclusions.</p>
<p>We now perform the basic ASCETIC inference for this second
scenario.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">resExamplePhylogeniesDataset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/asceticPhylogenies.html">asceticPhylogenies</a></span><span class="op">(</span></span>
<span>                                      dataset <span class="op">=</span> <span class="va">datasetExamplePhylogenies</span>,</span>
<span>                                      models <span class="op">=</span> <span class="va">modelsPhylogenies</span>,</span>
<span>                                      regularization <span class="op">=</span> <span class="st">"aic"</span>,</span>
<span>                                      command <span class="op">=</span> <span class="st">"hc"</span>,</span>
<span>                                      restarts <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p>The output of this analysis is a a list of 4 elements: 1) dataset,
input dataset; 2) models, input mutational trees; 3) poset, partially
order set among mutations estimated by ASCETIC from the agony ranking
and 4) inference, inferred ASCETIC evolutionary model for each selected
regularization.</p>
<p>The resulting poset and inference elements have the same semantics as
the previous example.</p>
<p>Also in this case we provide a method to improve the statistical
stability of the framework expecially in the situation of noisy data.
ASCETIC implements a bootstrap procedure which builds on the function
just described. This procedure should be preferred on real data. It can
be executed as follows.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">resExamplePhylogeniesDatasetBootstrap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/asceticPhylogeniesBootstrap.html">asceticPhylogeniesBootstrap</a></span><span class="op">(</span></span>
<span>                                                   dataset <span class="op">=</span> <span class="va">datasetExamplePhylogenies</span>,</span>
<span>                                                   models <span class="op">=</span> <span class="va">modelsPhylogenies</span>,</span>
<span>                                                   nsampling <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                                                   regularization <span class="op">=</span> <span class="st">"aic"</span>,</span>
<span>                                                   command <span class="op">=</span> <span class="st">"hc"</span>,</span>
<span>                                                   restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 0 </span></span>
<span><span class="co">## 0.2 </span></span>
<span><span class="co">## 0.4 </span></span>
<span><span class="co">## 0.6 </span></span>
<span><span class="co">## 0.8 </span></span>
<span><span class="co">## 1</span></span></code></pre>
<p>Finally, ASCETIC can perform the extraction of evolutionary
signatures significantly associated to prognosis and use them to
stratify patients. This can be done as follows.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">amlExample</span><span class="op">)</span></span>
<span><span class="va">resExampleEvosigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evosigs.html">evoSigs</a></span><span class="op">(</span>survivalData <span class="op">=</span> <span class="va">amlExample</span><span class="op">$</span><span class="va">survival_data</span>, </span>
<span>                             evolutionarySteps <span class="op">=</span> <span class="va">amlExample</span><span class="op">$</span><span class="va">evolutionary_steps</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">resExampleEvosigs</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "survivalAnalysis"   "evolutionarySteps"  "clustersPrevalence"</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">resExampleEvosigs</span><span class="op">$</span><span class="va">evolutionarySteps</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  ASXL1.to.NRAS DNMT3A.to.NRAS  DNMT3A.to.WT1 IDH1.to.PTPN11   IDH1.to.TET2 </span></span>
<span><span class="co">##     0.56617338    -0.26258331     1.07489799    -1.57548330    -1.50253436 </span></span>
<span><span class="co">##   IDH2.to.TET2   JAK2.to.NRAS  JAK2.to.RUNX1   NPM1.to.FLT3    Root.to.CBL </span></span>
<span><span class="co">##    -0.47637487     0.39346249     1.36329173     0.35692492     1.22736994 </span></span>
<span><span class="co">##   Root.to.IDH1   Root.to.JAK2    Root.to.KIT   Root.to.NPM1   Root.to.PHF6 </span></span>
<span><span class="co">##     0.51762715    -0.24124407    -0.76511252    -0.04687322     1.16513495 </span></span>
<span><span class="co">##  Root.to.PPM1D  Root.to.RAD21  Root.to.STAG2   Root.to.TP53  Root.to.U2AF1 </span></span>
<span><span class="co">##    -1.42425908    -0.40917150     0.48307602     1.52174568     0.69844295</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">resExampleEvosigs</span><span class="op">$</span><span class="va">clustersPrevalence</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    ASXL1.to.NRAS CALR.to.ASXL1 DNMT3A.to.NRAS DNMT3A.to.WT1 IDH1.to.DNMT3A</span></span>
<span><span class="co">## C1    0.00000000             0     0.11111111    0.00000000     0.00000000</span></span>
<span><span class="co">## C2    0.00000000             0     0.00000000    0.00000000     0.00000000</span></span>
<span><span class="co">## C3    0.04301075             0     0.02150538    0.08602151     0.08602151</span></span>
<span><span class="co">##    IDH1.to.PTPN11 IDH1.to.TET2 IDH2.to.TET2 JAK2.to.NRAS JAK2.to.RUNX1</span></span>
<span><span class="co">## C1     0.01587302   0.04761905   0.03174603  0.000000000    0.00000000</span></span>
<span><span class="co">## C2     0.00000000   0.00000000   0.00000000  0.004761905    0.00000000</span></span>
<span><span class="co">## C3     0.00000000   0.00000000   0.00000000  0.010752688    0.05376344</span></span>
<span><span class="co">##    NPM1.to.FLT3 Root.to.BRAF Root.to.CALR Root.to.CBL Root.to.IDH1 Root.to.IDH2</span></span>
<span><span class="co">## C1   0.04761905            0            0  0.00000000   0.03174603   0.04761905</span></span>
<span><span class="co">## C2   0.22380952            0            0  0.00000000   0.00000000   0.13333333</span></span>
<span><span class="co">## C3   0.01075269            0            0  0.01075269   0.16129032   0.01075269</span></span>
<span><span class="co">##    Root.to.JAK2 Root.to.KIT Root.to.NPM1 Root.to.PHF6 Root.to.PPM1D</span></span>
<span><span class="co">## C1  0.111111111  0.07936508   0.49206349   0.00000000    0.03174603</span></span>
<span><span class="co">## C2  0.004761905  0.00000000   0.20952381   0.00000000    0.00000000</span></span>
<span><span class="co">## C3  0.043010753  0.00000000   0.04301075   0.04301075    0.00000000</span></span>
<span><span class="co">##    Root.to.RAD21 Root.to.SF3B1 Root.to.STAG2 Root.to.TP53 Root.to.U2AF1</span></span>
<span><span class="co">## C1     0.1746032    0.00000000    0.00000000    0.0000000     0.0000000</span></span>
<span><span class="co">## C2     0.0000000    0.07619048    0.09047619    0.0000000     0.0000000</span></span>
<span><span class="co">## C3     0.0000000    0.01075269    0.03225806    0.3655914     0.2258065</span></span>
<span><span class="co">##    U2AF1.to.FLT3</span></span>
<span><span class="co">## C1    0.00000000</span></span>
<span><span class="co">## C2    0.00000000</span></span>
<span><span class="co">## C3    0.03225806</span></span></code></pre>
<p>The accumulation of alterations in driver genes can follow repeated
routes in different cancer patients. Detecting such trajectories could
be important in order to implement appropriate therapeutic responses. As
a matter of fact, being able to stratify cancer patients based on their
molecular evolution could enable the prediction of the future steps of
the disease progression, potentially allowing the execution of optimal
and personalized treatments that anticipate the next stages of the
cancer’s evolution. In our example, ASCETIC could identify a set of
evolutionary steps (see evolutionarySteps and clustersPrevalence
variables) with a significant association to prognosis.</p>
<p>Specifically, the evolutionarySteps variable reports 20 evolutionary
steps that, in this example dataset, show a significant association with
overall survival. The steps with negative values are negatively
associated with risk, meaning that patients with these mutations have a
better prognosis. On the other hand, steps with positive values increase
the risk, indicating that patients with these mutations have a worse
prognosis.</p>
<p>Finally, the clustersPrevalence variable provides the three clusters
discovered based on the detected evolutionary signatures, along with the
relative prevalences of each evolutionary step in the clusters.</p>
<p>We refer to the manual for a detailed description of each parameter
and to the ASCETIC manuscript for details on the method.</p>
</div>
<div class="section level2">
<h2 id="current-r-session">Current R Session<a class="anchor" aria-label="anchor" href="#current-r-session"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] ASCETIC_0.99.20  BiocStyle_2.30.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] sass_0.4.7          utf8_1.2.4          generics_0.1.3     </span></span>
<span><span class="co">##  [4] shape_1.4.6         stringi_1.7.12      lattice_0.22-5     </span></span>
<span><span class="co">##  [7] digest_0.6.33       magrittr_2.0.3      evaluate_0.22      </span></span>
<span><span class="co">## [10] grid_4.3.1          iterators_1.0.14    bookdown_0.36      </span></span>
<span><span class="co">## [13] fastmap_1.1.1       foreach_1.5.2       rprojroot_2.0.3    </span></span>
<span><span class="co">## [16] jsonlite_1.8.7      glmnet_4.1-8        Matrix_1.6-1.1     </span></span>
<span><span class="co">## [19] survival_3.5-7      BiocManager_1.30.22 purrr_1.0.2        </span></span>
<span><span class="co">## [22] fansi_1.0.5         scales_1.2.1        codetools_0.2-19   </span></span>
<span><span class="co">## [25] textshaping_0.3.7   jquerylib_0.1.4     cli_3.6.1          </span></span>
<span><span class="co">## [28] rlang_1.1.1         bnlearn_4.9         splines_4.3.1      </span></span>
<span><span class="co">## [31] munsell_0.5.0       cachem_1.0.8        yaml_2.3.7         </span></span>
<span><span class="co">## [34] tools_4.3.1         parallel_4.3.1      memoise_2.0.1      </span></span>
<span><span class="co">## [37] dplyr_1.1.3         colorspace_2.1-0    ggplot2_3.4.4      </span></span>
<span><span class="co">## [40] vctrs_0.6.4         R6_2.5.1            lifecycle_1.0.3    </span></span>
<span><span class="co">## [43] stringr_1.5.0       fs_1.6.3            ragg_1.2.6         </span></span>
<span><span class="co">## [46] pkgconfig_2.0.3     desc_1.4.2          pkgdown_2.0.7.9000 </span></span>
<span><span class="co">## [49] bslib_0.5.1         pillar_1.9.0        gtable_0.3.4       </span></span>
<span><span class="co">## [52] glue_1.6.2          Rcpp_1.0.11         systemfonts_1.0.5  </span></span>
<span><span class="co">## [55] xfun_0.40           tibble_3.2.1        tidyselect_1.2.0   </span></span>
<span><span class="co">## [58] knitr_1.45          htmltools_0.5.6.1   rmarkdown_2.25     </span></span>
<span><span class="co">## [61] compiler_4.3.1</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Daniele Ramazzotti, Luca De Sano.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
