<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>An introduction to the `ASCETIC` R package • ASCETIC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="An introduction to the `ASCETIC` R package">
<meta property="og:description" content="ASCETIC">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ASCETIC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/ASCETIC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/danro9685/ASCETIC/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>An introduction to the <code>ASCETIC</code> R
package</h1>
                        <h4 data-toc-skip class="author">Luca De
Sano</h4>
                        <h4 data-toc-skip class="author">Daniele
Ramazzotti</h4>
            
            <h4 data-toc-skip class="date">January 14, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/danro9685/ASCETIC/blob/HEAD/vignettes/ASCETIC.rmd" class="external-link"><code>vignettes/ASCETIC.rmd</code></a></small>
      <div class="hidden name"><code>ASCETIC.rmd</code></div>

    </div>

    
    
<p>Cancer development is a stochastic process involving large
populations of cells. Random genetic and epigenetic alterations commonly
occurring in any cell can occasionally be beneficial to neoplastic ones,
thus defining clones characterized by a functional selective advantage.
During clonal evolution, certain clones can be positively selected for
increased proliferation and survival ability, outgrowing competing cells
and this can eventually lead to invasion and metastasis. Throughout such
a multi-step stochastic process, cancer cells can acquire over time a
set of biological capabilities that sometimes are referred to as
hallmarks. Not all variants are involved in their acquisition, but only
a relative small subset of them – i.e., the drivers –, while most
mutations present in the cancer clones do not increase their fitness –
i.e., the passengers.</p>
<p>ASCETIC (Agony-baSed Cancer EvoluTion InferenCe) is a novel framework
for the inference of a set of statistically significant temporal
patterns involving alternations in driver genes from cancer genomics
data. The framework can be applied to data at different resolutions,
from classical NGS genomics data of cancer samples, to single-cell
data.</p>
<p>In its basic implementation, ASCETIC requires two main inputs: (i) a
binary matrix where rows are patients (i.e., samples) and columns are
mutations. Each cell of the matrix is 1 if the related mutation was
observed in the sample; 0 otherwise. (2) Information of temporal patters
across the considered driver genes. Such information can be determined
either from classical NGS genomics data considering a single biopsy per
patient, in terms of cancer cell fractions reported for each gene, or
from data at different resolutions such as multi-region or single-cell
data where multiple samples per patients are provided; in this latter
case, ASCETIC takes as input a phylogenetic tree per patient.</p>
<p>In this vignette, we give an overview of the package by presenting
some of its main functions.</p>
<div class="section level2">
<h2 id="installing-the-ascetic-r-package">Installing the ASCETIC R package<a class="anchor" aria-label="anchor" href="#installing-the-ascetic-r-package"></a>
</h2>
<p>The ASCETIC package can be installed from Bioconductor as
follows.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"ASCETIC"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="changelog">Changelog<a class="anchor" aria-label="anchor" href="#changelog"></a>
</h2>
<ul>
<li>
<em>1.0.0</em> Package released in October 2022.</li>
</ul>
</div>
<div class="section level2">
<h2 id="using-the-ascetic-r-package">Using the ASCETIC R package<a class="anchor" aria-label="anchor" href="#using-the-ascetic-r-package"></a>
</h2>
<p>We provide within the package example datasets for both the single
biopsy and the multi-regions/single-cell cases. We now load such example
datasets.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/danro9685/ASCETIC" class="external-link">"ASCETIC"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dataset_example_single_samples</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ccf_dataset_example_single_samples</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">dataset_example_phylogenies</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">models_phylogenies</span><span class="op">)</span></span></code></pre></div>
<p>First, we outline the inference steps for the single biopsy NGS data
scenario, where only one sample per patient is available and the
temporal patters among genes are estimated considering cancer cell
fractions data. To test this, we consider an example dataset comprising
10 hepatocellular carcinoma sequencing samples from Nguyen, Bastien, et
al., 2022. We recall that such (reduced and partial) dataset was
provided only for computational testing and should not be used for
biological analysis/conclusions.</p>
<p>We now perform the basic ASCETIC inference.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">res_example_single_samples</span> <span class="op">=</span> <span class="fu"><a href="../reference/perform.ascetic.ccf.dataset.html">perform.ascetic.ccf.dataset</a></span><span class="op">(</span> </span>
<span>                                dataset <span class="op">=</span> <span class="va">dataset_example_single_samples</span>, </span>
<span>                                ccf.dataset <span class="op">=</span> <span class="va">ccf_dataset_example_single_samples</span>, </span>
<span>                                regularization <span class="op">=</span> <span class="st">"aic"</span>, </span>
<span>                                command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                                restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<p>The output of this analysis is a a list of 4 elements: 1) dataset,
input dataset; 2) ccf dataset, input ccf dataset; 3) poset, partially
order set among mutations estimated by ASCETIC from the agony ranking
and 4) inference, inferred ASCETIC evolutionary model for each selected
regularization.</p>
<p>To improve the statistical stability of the framework expecially in
the common situation of noisy data, ASCETIC implements a re-sampling
algorithm which builds on the function just described. This procedure is
more computational intensive compared to the basic one, but should be
preferred on real data. It requires an additional dataset with variant
allele frequencies information and can be executed as follows.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">vaf_dataset_example_single_samples</span><span class="op">)</span></span>
<span><span class="va">res_example_single_samples_resampling</span> <span class="op">=</span> <span class="fu"><a href="../reference/perform.ascetic.ccf.dataset.resampling.html">perform.ascetic.ccf.dataset.resampling</a></span><span class="op">(</span> </span>
<span>                                            dataset <span class="op">=</span> <span class="va">dataset_example_single_samples</span>, </span>
<span>                                            ccf.dataset <span class="op">=</span> <span class="va">ccf_dataset_example_single_samples</span>, </span>
<span>                                            vaf.dataset <span class="op">=</span> <span class="va">vaf_dataset_example_single_samples</span>, </span>
<span>                                            nsampling <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                                            regularization <span class="op">=</span> <span class="st">"aic"</span>, </span>
<span>                                            command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                                            restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 0 </span></span>
<span><span class="co">## 0.2 </span></span>
<span><span class="co">## 0.4 </span></span>
<span><span class="co">## 0.6 </span></span>
<span><span class="co">## 0.8 </span></span>
<span><span class="co">## 1</span></span></code></pre>
<p>We refer to the manual for a detailed description of each
parameter.</p>
<p>We now describe the execution of the ASCETIC framework when
high-resolution cancer genomics data are available, such as
multi-regions data or single-cell data. In this case, phylogenetic
reconstructions for each patient are available and are given as input to
ASCETIC. We consider an other example dataset consisting of 10 myeloid
malignancies samples sequenced with Tapestri from Miles, Linde A., et
al., 2020. We recall that also in this case, such (reduced and partial)
dataset was provided only for computational testing and should not be
used for biological analysis/conclusions.</p>
<p>We now perform the basic ASCETIC inference for this second
scenario.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">res_example_phylogenies_dataset</span> <span class="op">=</span> <span class="fu"><a href="../reference/perform.ascetic.phylogenies.dataset.html">perform.ascetic.phylogenies.dataset</a></span><span class="op">(</span> </span>
<span>                                    dataset <span class="op">=</span> <span class="va">dataset_example_phylogenies</span>, </span>
<span>                                    models <span class="op">=</span> <span class="va">models_phylogenies</span>, </span>
<span>                                    regularization <span class="op">=</span> <span class="st">"aic"</span>, </span>
<span>                                    command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                                    restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<p>The output of this analysis is a a list of 4 elements: 1) dataset,
input dataset; 2) models, input phylogenetic models; 3) poset, partially
order set among mutations estimated by ASCETIC from the agony ranking
and 4) inference, inferred ASCETIC evolutionary model for each selected
regularization.</p>
<p>Also in this case we provide a method to improve the statistical
stability of the framework expecially in the situation of noisy data.
ASCETIC implements a bootstrap procedure which builds on the function
just described. This procedure should be preferred on real data. It can
be executed as follows.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="va">res_example_phylogenies_dataset_bootstrap</span> <span class="op">=</span> <span class="fu"><a href="../reference/perform.ascetic.phylogenies.dataset.bootstrap.html">perform.ascetic.phylogenies.dataset.bootstrap</a></span><span class="op">(</span> </span>
<span>                                                dataset <span class="op">=</span> <span class="va">dataset_example_phylogenies</span>, </span>
<span>                                                models <span class="op">=</span> <span class="va">models_phylogenies</span>, </span>
<span>                                                nsampling <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                                                regularization <span class="op">=</span> <span class="st">"aic"</span>, </span>
<span>                                                command <span class="op">=</span> <span class="st">"hc"</span>, </span>
<span>                                                restarts <span class="op">=</span> <span class="fl">0</span> <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 0 </span></span>
<span><span class="co">## 0.2 </span></span>
<span><span class="co">## 0.4 </span></span>
<span><span class="co">## 0.6 </span></span>
<span><span class="co">## 0.8 </span></span>
<span><span class="co">## 1</span></span></code></pre>
<p>Finally, ASCETIC can perform the extraction of evolutionary
signatures significantly associated to prognosis and use them to
stratify patients. This can be done as follows.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">aml_example</span><span class="op">)</span></span>
<span><span class="va">res_example_evosigs</span> <span class="op">=</span> <span class="fu"><a href="../reference/evosigs.html">evosigs</a></span><span class="op">(</span> </span>
<span>                              survival_data <span class="op">=</span> <span class="va">aml_example</span><span class="op">$</span><span class="va">survival_data</span>, </span>
<span>                              evolutionary_steps <span class="op">=</span> <span class="va">aml_example</span><span class="op">$</span><span class="va">evolutionary_steps</span> <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">res_example_evosigs</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "survival_analysis"   "evolutionary_steps"  "clusters_prevalence"</span></span></code></pre>
<p>We refer to the manual for a detailed description of each parameter
and to the ASCETIC manuscript for details on the method.</p>
</div>
<div class="section level2">
<h2 id="current-r-session">Current R Session<a class="anchor" aria-label="anchor" href="#current-r-session"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] ASCETIC_0.99.7   BiocStyle_2.26.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] Rcpp_1.0.9          pillar_1.8.1        bslib_0.4.2        </span></span>
<span><span class="co">##  [4] compiler_4.2.2      BiocManager_1.30.19 jquerylib_0.1.4    </span></span>
<span><span class="co">##  [7] iterators_1.0.14    tools_4.2.2         digest_0.6.31      </span></span>
<span><span class="co">## [10] lattice_0.20-45     tibble_3.1.8        jsonlite_1.8.4     </span></span>
<span><span class="co">## [13] evaluate_0.19       memoise_2.0.1       lifecycle_1.0.3    </span></span>
<span><span class="co">## [16] gtable_0.3.1        pkgconfig_2.0.3     rlang_1.0.6        </span></span>
<span><span class="co">## [19] foreach_1.5.2       Matrix_1.5-1        cli_3.6.0          </span></span>
<span><span class="co">## [22] yaml_2.3.6          parallel_4.2.2      pkgdown_2.0.7.9000 </span></span>
<span><span class="co">## [25] xfun_0.36           fastmap_1.1.0       stringr_1.5.0      </span></span>
<span><span class="co">## [28] knitr_1.41          desc_1.4.2          fs_1.5.2           </span></span>
<span><span class="co">## [31] sass_0.4.4          vctrs_0.5.1         systemfonts_1.0.4  </span></span>
<span><span class="co">## [34] glmnet_4.1-6        rprojroot_2.0.3     grid_4.2.2         </span></span>
<span><span class="co">## [37] glue_1.6.2          R6_2.5.1            textshaping_0.3.6  </span></span>
<span><span class="co">## [40] fansi_1.0.3         survival_3.4-0      bnlearn_4.8.1      </span></span>
<span><span class="co">## [43] rmarkdown_2.19      bookdown_0.31       purrr_1.0.1        </span></span>
<span><span class="co">## [46] ggplot2_3.4.0       magrittr_2.0.3      splines_4.2.2      </span></span>
<span><span class="co">## [49] codetools_0.2-18    scales_1.2.1        htmltools_0.5.4    </span></span>
<span><span class="co">## [52] shape_1.4.6         colorspace_2.0-3    ragg_1.2.5         </span></span>
<span><span class="co">## [55] utf8_1.2.2          stringi_1.7.12      munsell_0.5.0      </span></span>
<span><span class="co">## [58] cachem_1.0.6</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Daniele Ramazzotti, Luca De Sano.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
